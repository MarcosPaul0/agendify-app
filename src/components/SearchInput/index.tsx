import { MagnifyingGlass } from 'phosphor-react-native';
import { View, TextInput } from 'react-native';
import { COLORS } from '@constants/colors.constant';
import { ISearchInputProps } from './interfaces/searchInputProps.interface';

export function SearchInput({
  search,
  changeSearch,
  ...rest
}: ISearchInputProps) {
  return (
    <View
      className={`
        h-11 pl-2.5 my-4 flex-row items-center mx-9
        bg-GRAY_100 border border-GRAY_500 rounded-lg
      `}
    >
      <MagnifyingGlass size={24} color={COLORS.GRAY_700} weight="bold" />
      <TextInput
        className={`
          text-base text-GRAY_800 w-full h-11 pl-2.5 
        `}
        value={search}
        onChangeText={changeSearch}
        selectionColor={COLORS.GRAY_800}
        {...rest}
      />
    </View>
  );
}
