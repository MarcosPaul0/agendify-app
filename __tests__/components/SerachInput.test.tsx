import { render } from '@testing-library/react-native';
import { SearchInput } from '@components/SearchInput';

describe('Search Input', () => {
  it('Match with snapshot', async () => {
    const tree = render(
      <SearchInput search="fake-search" changeSearch={() => null} />
    );

    expect(tree).toMatchSnapshot();
  });
});
