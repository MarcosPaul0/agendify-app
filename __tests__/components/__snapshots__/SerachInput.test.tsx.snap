// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input Match with snapshot 1`] = `
<View
  style={
    [
      {
        "height": 44,
      },
      {
        "paddingLeft": 10,
      },
      {
        "marginBottom": 16,
        "marginTop": 16,
      },
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "marginLeft": 36,
        "marginRight": 36,
      },
      {
        "backgroundColor": "#F5F5F5",
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "borderBottomColor": "#ADB5BD",
        "borderLeftColor": "#ADB5BD",
        "borderRightColor": "#ADB5BD",
        "borderTopColor": "#ADB5BD",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    color="#495057"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={24}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    tintColor="#495057"
    vbHeight={256}
    vbWidth={256}
    width={24}
  >
    <RNSVGGroup>
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGCircle
        cx="116"
        cy="116"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        r="84"
        stroke={
          {
            "payload": 4282994775,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="24"
      />
      <RNSVGLine
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282994775,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="24"
        x1="175.4"
        x2="224"
        y1="175.4"
        y2="224"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <TextInput
    onChangeText={[Function]}
    selectionColor="#343A40"
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#343A40",
        },
        {
          "width": "100%",
        },
        {
          "height": 44,
        },
        {
          "paddingLeft": 10,
        },
      ]
    }
    value="fake-search"
  />
</View>
`;
